core/src/main/java/com/graphhopper/routing/ch/PrepareContractionHierarchies.java

```diff
@@ -320,6 +320,10 @@ void contractNodes() {
 
                 prepareGraph.disconnect(vehicleAllTmpExplorer, iter);
             }
+
+            if(Thread.currentThread().isInterrupted()){
```

Comment:
Ok, I moved it to the inner while. I also measured the performance, which has not decreased, see the updated table above.